### PR TITLE
Window Features: minor corrections to which processes can set foreground

### PR DESCRIPTION
**Pull request by metathinker**
The conditions given for when a process can set the foreground window with SetForegroundWindow were not quite right; I've updated them accordingly and made some other minor rewordings.

Pairs with https://github.com/MicrosoftDocs/sdk-api/pull/394

### CODE REVIEW

**Comment by PRMerger8** [comment]
@metathinker : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: desktop-src/winmsg/window-features.md [hunk 1]
```diff
@@ -162,18 +162,35 @@ The user sets the foreground window by clicking a window, or by using the ALT+TA
...
--   The foreground is not locked (see [**LockSetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-locksetforegroundwindow)).
--   The foreground lock time-out has expired (see **SPI\_GETFOREGROUNDLOCKTIMEOUT** in [**SystemParametersInfo**](/windows/win32/api/winuser/nf-winuser-systemparametersinfoa)).
--   No menus are active.
-
-A process that can set the foreground window can enable another process to set the foreground window by calling the [**AllowSetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-allowsetforegroundwindow) function, or by calling the [**BroadcastSystemMessage**](/windows/win32/api/winuser/nf-winuser-broadcastsystemmessage) function with the **BSF\_ALLOWSFW** flag. The foreground process can disable calls to [**SetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-setforegroundwindow) by calling the [**LockSetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-locksetforegroundwindow) function.
+The system restricts which applications can set the foreground window.
+A process can set the foreground window by calling [**SetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-setforegroundwindow) only if:
```
Comment (hickeys) [inline desktop-src/winmsg/window-features.md]: Please revert this change, the previous line already covers setting the foreground window via `SetForegroundWindow`

Also, please do not add carriage returns between sentences. Single carriage returns do not affect the final published page, and they break Git's diffing tools.

File: desktop-src/winmsg/window-features.md [hunk 2]
```diff
@@ -162,18 +162,35 @@ The user sets the foreground window by clicking a window, or by using the ALT+TA
...
+
+- At least one of the following conditions is true:
+  - The calling process is the foreground process -
+    the process that created the previous foreground thread
+    and foreground window.
+  - The calling process was started by the foreground process.
+  - There is currently no foreground window, and thus no foreground process.
+  - The calling process received the last input event.
+  - Either the foreground process or the calling process is being debugged.
+- And all of the following conditions are true:
```
Comment (hickeys) [inline desktop-src/winmsg/window-features.md]: Please restructure this section so that the "All of the following" comes first, followed by "Additionally, at least one of the following"

File: desktop-src/winmsg/window-features.md [hunk 3]
```diff
@@ -162,18 +162,35 @@ The user sets the foreground window by clicking a window, or by using the ALT+TA
...
--   The foreground lock time-out has expired (see **SPI\_GETFOREGROUNDLOCKTIMEOUT** in [**SystemParametersInfo**](/windows/win32/api/winuser/nf-winuser-systemparametersinfoa)).
--   No menus are active.
-
-A process that can set the foreground window can enable another process to set the foreground window by calling the [**AllowSetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-allowsetforegroundwindow) function, or by calling the [**BroadcastSystemMessage**](/windows/win32/api/winuser/nf-winuser-broadcastsystemmessage) function with the **BSF\_ALLOWSFW** flag. The foreground process can disable calls to [**SetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-setforegroundwindow) by calling the [**LockSetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-locksetforegroundwindow) function.
+The system restricts which applications can set the foreground window.
+A process can set the foreground window by calling [**SetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-setforegroundwindow) only if:
+
+- At least one of the following conditions is true:
+  - The calling process is the foreground process -
```
Comment (hickeys) [inline desktop-src/winmsg/window-features.md]: Please remove the definition of foreground process from this item

File: desktop-src/winmsg/window-features.md [hunk 4]
```diff
@@ -162,18 +162,35 @@ The user sets the foreground window by clicking a window, or by using the ALT+TA
...
+    the process that created the previous foreground thread
+    and foreground window.
+  - The calling process was started by the foreground process.
+  - There is currently no foreground window, and thus no foreground process.
+  - The calling process received the last input event.
+  - Either the foreground process or the calling process is being debugged.
+- And all of the following conditions are true:
+  - The calling process belongs to a desktop application,
```
Comment (hickeys) [inline desktop-src/winmsg/window-features.md]: Please remove this item - adding a functional change like this requires more verification and testing.

File: desktop-src/winmsg/window-features.md [hunk 5]
```diff
@@ -162,18 +162,35 @@ The user sets the foreground window by clicking a window, or by using the ALT+TA
...
+  - The foreground process has not disabled calls to [**SetForegroundWindow**](
+    /windows/win32/api/winuser/nf-winuser-setforegroundwindow) by a previous call to the
+    [**LockSetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-locksetforegroundwindow)
+    function.
+  - The foreground lock time-out has expired (see
+    [**SPI_GETFOREGROUNDLOCKTIMEOUT** in **SystemParametersInfo**](
+    /windows/win32/api/winuser/nf-winuser-systemparametersinfoa#SPI_GETFOREGROUNDLOCKTIMEOUT)).
+  - No menus are active.
+
+It is possible for a process to be denied the right to set the foreground window
```
Comment (hickeys) [inline desktop-src/winmsg/window-features.md]: Please revert all whitespace changes past this point

**Review by hickeys — CHANGES_REQUESTED** [review]
Thanks for the PR. I've requested a few changes, but overall I like your edits.

**Comment by PRMerger19** [comment]
@metathinker : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: desktop-src/winmsg/window-features.md (hunk 4 above)
Comment (metathinker) [reply desktop-src/winmsg/window-features.md]: This is a copy and correction (foreground process --> calling process) of this clause that already exists in the Remarks section for [SetForegroundWindow()](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setforegroundwindow)'s doc page:

> The foreground process is not a Modern Application or the Start Screen.

Assuming we'd like to keep this conceptual topic in sync with those API reference pages, there are a few things we can do here. Here are some ideas:

1. Keep my current addition here.
1. Delete the added clause from here, and from SetForegroundWindow too. (I can do that as part of my companion PR https://github.com/MicrosoftDocs/sdk-api/pull/394 .)
1. Keep my current addition here, and then change the pages for SetForegroundWindow and [AllowSetForegroundWindow()](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-allowsetforegroundwindow) to remove the "who can set foreground?" remarks entirely and link to the discussion here.
1. Delete the added clause here, and then remove "who can set foreground?" discussion from the API function pages like in the previous option.

What would you suggest? Of my 4 ideas, I'm leaning toward the last one.

Comment (hickeys) [reply desktop-src/winmsg/window-features.md]: @metathinker I appreciate your points, but I'd feel more comfortable if I could get confirmation from a current MSFT employee. I see that you used to work here - is there you recommend I speak to for confirmation?

Comment (metathinker) [reply desktop-src/winmsg/window-features.md]: @hickeys Check the ntuser@microsoft.com alias if it's still around. Otherwise, I forget who exactly owns user32.dll (Alex Snitkovsky?), but you should be able to look them up on the ownership system (https://osgwiki.com/wiki/Ownership ?).

**Comment by hickeys** [comment]
Thanks @metathinker 

The DL isn't in use anymore, but I've reached out to Alex. I'll keep you posted when I hear back.

**Review by hickeys — APPROVED** [review]
Alex approved the changes
